autopts/client.py

```diff
@@ -1375,7 +1375,7 @@ def sigint_handler(sig, frame):
             return self.main(args)
         except BaseException as e:
             if not isinstance(e, KeyboardInterrupt):   # Ctrl-C
-                if e.code != 0:
+                if getattr(e, "code", 1) != 0:
                     # Exit with traceback
                     logging.exception(e)
```

Comment:
We should consider doing this in a more pythonic way IMO. 

So it should be of the form of
```
try: 
    # do stuff
except KeyboardInterrupt:
    # no-op
    pass
except SystemExit as e:
    if e.code != 0:
        # Exit with traceback
        logging.exception(e)
except TimeoutError as e:
    logging.exception(e)
finally:
    set_global_end()
    self.cleanup()
    raise
```

Generally catching `BaseException` in Python is frowned upon for the reasons here. It catches everything, and makes it harder to determine what errors may happen here and how they are handled.